--- startnoui.c	2017-07-22 08:00:39.126236006 +0300
+++ startnoui.c	2023-11-02 14:20:43.522000000 +0200
@@ -106,8 +106,8 @@
     bindtextdomain("FontForge", getLocaleDir());
     textdomain("FontForge");
 
-    if ( default_encoding==NULL )
-	default_encoding=FindOrMakeEncoding("ISO8859-1");
+    //if ( default_encoding==NULL )
+	//default_encoding=FindOrMakeEncoding("ISO8859-1");
     if ( default_encoding==NULL )
 	default_encoding=&custom;	/* In case iconv is broken */
     CheckIsScript(argc,argv);		/* Will run the script and exit if it is a script */
