--- libffi.map.in	2023-10-30 05:05:56.339000000 +0200
+++ libffi.map.in	2023-10-30 05:18:28.143000000 +0200
@@ -20,7 +20,9 @@
 	ffi_type_sint64;
 	ffi_type_float;
 	ffi_type_double;
+#if !defined(__arm__) && !defined(__aarch64__) && !defined(__i386__)
 	ffi_type_longdouble;
+#endif
 	ffi_type_pointer;
 
 	/* Exported functions.  */
@@ -52,7 +54,9 @@
 	/* Exported data variables.  */
 	ffi_type_complex_float;
 	ffi_type_complex_double;
+#if !defined(__arm__) && !defined(__aarch64__) && !defined(__i386__)
 	ffi_type_complex_longdouble;
+#endif
 } LIBFFI_BASE_8.0;
 #endif
 
