--- config.h.in	2017-07-31 11:31:43.570015000 +0300
+++ config.h.in	2023-11-03 06:04:58.549000000 +0200
@@ -1514,8 +1514,10 @@
 /* Enable large inode numbers on Mac OS X 10.5. */
 #undef _DARWIN_USE_64_BIT_INODE
 
+#if __ANDROID_MIN_SDK_VERSION__ >= 24
 /* Number of bits in a file offset, on hosts where this is settable. */
 #undef _FILE_OFFSET_BITS
+#endif
 
 /* Platform specific stuff */
 #undef _Keyboard
