--- CMakeLists.txt	2023-10-02 22:55:54.000000000 +0300
+++ CMakeLists.txt	2023-11-03 13:17:53.216000000 +0200
@@ -223,7 +223,7 @@
 find_soft_mandatory_package(ENABLE_QT6 Qt6Test ${QT6_VERSION})
 
 # Check for Cairo rendering backend
-macro_optional_find_package(Cairo ${CAIRO_VERSION})
+pkg_search_module(CAIRO REQUIRED IMPORTED_TARGET cairo)
 
 find_package(Boost 1.71.0)
 if(Boost_FOUND)
@@ -233,6 +233,7 @@
 endif()
 
 if(CAIRO_FOUND)
+  set(CAIRO_LIBRARIES PkgConfig::CAIRO)
   set(HAVE_CAIRO ${CAIRO_FOUND})
   set(CAIRO_FEATURE "#define POPPLER_HAS_CAIRO 1")
   set(CAIRO_REQ "cairo")
