--- ConfigureChecks.cmake	2021-02-01 23:23:11.000000000 +0200
+++ ConfigureChecks.cmake	2023-11-03 13:03:23.673000000 +0200
@@ -24,7 +24,7 @@
 check_include_files(stdlib.h HAVE_STDLIB_H)
 check_include_files(sys/mman.h HAVE_SYS_MMAN_H)
 check_include_files(sys/stat.h HAVE_SYS_STAT_H)
-check_include_files(unistd.h HAVE_UNISTD_H)
+SET(HAVE_UNISTD_H 1)
 
 check_function_exists(fseek64 HAVE_FSEEK64)
 check_symbol_exists(fseeko "stdio.h" HAVE_FSEEKO)
