--- fontforge/CMakeLists.txt	2020-01-27 10:39:51.000000000 +0200
+++ fontforge/CMakeLists.txt	2023-11-03 10:04:11.517000000 +0200
@@ -112,8 +112,6 @@
 )
 
 add_library(fontforge
-  ${FONTFORGE_NOINST_HEADERS}
-  ${FONTFORGE_INST_HEADERS}
   activeinui.c
   asmfpst.c
   autohint.c
@@ -209,6 +207,9 @@
   zapfnomen.c
 )
 
+set_target_properties(fontforge PROPERTIES PUBLIC_HEADER "${FONTFORGE_INST_HEADERS}")
+set_target_properties(fontforge PROPERTIES PRIVATE_HEADER "${FONTFORGE_NOINST_HEADERS}")
+
 if(ENABLE_WOFF2_RESULT)
   target_sources(fontforge PRIVATE woff2.cc)
 endif()
@@ -269,11 +270,13 @@
   target_link_libraries(fontforge PRIVATE gunicode gutils)
 endif()
 
-# No dev package -> no need to install if static
-if(BUILD_SHARED_LIBS)
-  if(WIN32)
-    install(TARGETS fontforge RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
-  else()
-    install(TARGETS fontforge RUNTIME DESTINATION ${CMAKE_INSTALL_LIBDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
-  endif()
+if(WIN32)
+  install(TARGETS fontforge RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+else()
+  install(TARGETS fontforge
+    RUNTIME DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+  )
 endif()
