--- pdf2htmlEX/src/util/ffw.c  2023-12-14 07:45:44.994000000 +0200
+++ pdf2htmlEX/src/util/ffw.c  2023-12-14 07:46:21.870000000 +0200
@@ -70,8 +70,10 @@
 void ffw_init(const char* progPath, int debug)
 {
     ffwSetAction("initialize");
-    char *localProgPath = strdup(progPath);
-    FindProgDir(localProgPath);
+    char *localProgPath = NULL;
+    if (progPath != NULL)
+        localProgPath = strdup(progPath);
+    FindProgRoot(localProgPath);
     InitSimpleStuff();
     if ( default_encoding==NULL )
         default_encoding=FindOrMakeEncoding("ISO8859-1");
@@ -290,7 +290,7 @@
     ffwClearAction();
 }

-void ffw_reencode_raw(int32 * mapping, int mapping_len, int force)
+void ffw_reencode_raw(int32_t * mapping, int mapping_len, int force)
 {
     ffwSetAction("re-encode (raw1)");
     Encoding * enc = calloc(1, sizeof(Encoding));
@@ -574,7 +574,7 @@
 void ffw_override_fstype(void)
 {
     ffwSetAction("override the fstype of");
-    *(int16 *)(&cur_fv->sf->pfminfo.fstype) = 0;
+    *(int16_t *)(&cur_fv->sf->pfminfo.fstype) = 0;
     cur_fv->sf->pfminfo.pfmset = true;
     cur_fv->sf->changed = true;
     ffwClearAction();
