--- Make.inc	2022-01-20 01:43:52.000000000 +0200
+++ Make.inc	2023-12-13 04:04:04.386000000 +0200
@@ -148,11 +148,13 @@
 endif
 endif

-ifneq ($(filter $(ARCH),i387 amd64),)
+ifeq ($(ARCH), amd64)
 # Determines whether `long double` is the same as `double` on this arch.
 # linux x86_64, for instance, `long double` is 80 bits wide, whereas on macOS aarch64,
 # `long double` is the same as `double`.
 LONG_DOUBLE_NOT_DOUBLE := 1
+else ifeq ($(ARCH), i387)
+LONG_DOUBLE_NOT_DOUBLE := 0
 else ifeq ($(ARCH), aarch64)
 ifeq ($(filter $(OS),Darwin WINNT),)
 LONG_DOUBLE_NOT_DOUBLE := 1
