--- Makefile
+++ Makefile	2023-12-22 20:41:29.633000000 +0200
@@ -43,8 +43,8 @@
 OLM_MAJOR_MINOR_SHLIB_EXT := $(SOMAJOR).$(SOMINOR).$(SHLIB_EXT)
 OLM_MAJOR_SHLIB_EXT := $(SOMAJOR).$(SHLIB_EXT)
 else
-OLM_MAJOR_MINOR_SHLIB_EXT := $(SHLIB_EXT).$(SOMAJOR).$(SOMINOR)
-OLM_MAJOR_SHLIB_EXT := $(SHLIB_EXT).$(SOMAJOR)
+OLM_MAJOR_MINOR_SHLIB_EXT := $(SHLIB_EXT) #.$(SOMAJOR).$(SOMINOR)
+OLM_MAJOR_SHLIB_EXT := $(SHLIB_EXT) #.$(SOMAJOR)
 endif
 LDFLAGS_add += -Wl,$(SONAME_FLAG),libopenlibm.$(OLM_MAJOR_SHLIB_EXT)
 endif
@@ -69,10 +69,6 @@
 
 libopenlibm.$(OLM_MAJOR_MINOR_SHLIB_EXT): $(OBJS)
 	$(CC) -shared $(OBJS) $(LDFLAGS) $(LDFLAGS_add) -o $@
-ifneq ($(OS),WINNT)
-	ln -sf $@ libopenlibm.$(OLM_MAJOR_SHLIB_EXT)
-	ln -sf $@ libopenlibm.$(SHLIB_EXT)
-endif
 
 test/test-double: libopenlibm.$(OLM_MAJOR_MINOR_SHLIB_EXT)
 	$(MAKE) -C test test-double
