--- gio/giounix-private.c
+++ gio/giounix-private.c
@@ -25,6 +25,22 @@
 #include <sys/types.h>
 #if defined (HAVE_EPOLL_CREATE)
 #include <sys/epoll.h>
+#if __ANDROID_API__ < 21
+// Patch inspired by (taken from):
+// https://github.com/deltachat/deltachat-android/pull/2324
+// fcntl only if epoll created successfully
+#include <fcntl.h>
+#define EPOLL_CLOEXEC O_CLOEXEC
+static int epoll_create1(int flags) {
+  int fd = epoll_create(1);
+  if (-1 != fd && flags & O_CLOEXEC) {
+      int f = fcntl(fd, F_GETFD);
+      fcntl(fd, F_SETFD, f | FD_CLOEXEC);
+  }
+  return fd;
+}
+#endif
+                        
 #elif defined (HAVE_KQUEUE)
 #include <sys/event.h>
 #include <sys/time.h>
