--- CMakeLists.txt	2023-11-03 12:33:39.721000000 +0200
+++ CMakeLists.txt	2023-11-03 12:34:02.079000000 +0200
@@ -156,8 +156,9 @@
   endif()
 endif()
 
-macro_optional_find_package(Cairo ${CAIRO_VERSION})
+pkg_search_module(CAIRO REQUIRED IMPORTED_TARGET cairo)
 if(CAIRO_FOUND)
+  set(CAIRO_LIBRARIES PkgConfig::CAIRO)
   set(HAVE_CAIRO ${CAIRO_FOUND})
   set(CAIRO_FEATURE "#define POPPLER_HAS_CAIRO 1")
   set(CAIRO_REQ "cairo")
