--- ConfigureChecks.cmake	2019-09-22 18:56:22.000000000 +0300
+++ ConfigureChecks.cmake	2023-11-03 10:50:16.125000000 +0200
@@ -24,7 +24,7 @@
 check_include_files(stdlib.h HAVE_STDLIB_H)
 check_include_files(sys/mman.h HAVE_SYS_MMAN_H)
 check_include_files(sys/stat.h HAVE_SYS_STAT_H)
-check_include_files(unistd.h HAVE_UNISTD_H)
+SET(HAVE_UNISTD_H 1)
 check_include_file_cxx(codecvt HAVE_CODECVT)
 
 check_function_exists(fseek64 HAVE_FSEEK64)
