--- CMakeLists.txt	2019-09-27 15:43:55.000000000 +0300
+++ CMakeLists.txt	2023-11-04 09:56:08.477000000 +0200
@@ -46,15 +46,6 @@
 pkg_search_module(FONTFORGE REQUIRED IMPORTED_TARGET libfontforge)
 set(PDF2HTMLEX_LIBS ${PDF2HTMLEX_LIBS} PkgConfig::FONTFORGE)
 
-# debug build flags (overwrite default cmake debug flags)
-set(CMAKE_C_FLAGS_DEBUG "-ggdb -pg")
-set(CMAKE_CXX_FLAGS_DEBUG "-ggdb -pg")
-set(CMAKE_EXE_LINKER_FLAGS_DEBUG "-pg")
-
-# release build flags (overwrite default cmake release flags)
-set(CMAKE_C_FLAGS_RELEASE "-O2 -DNDEBUG")
-set(CMAKE_CXX_FLAGS_RELEASE "-O2 -DNDEBUG")
-
 # generic flags
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Woverloaded-virtual")
