--- cmake/packages/FindGLIB.cmake	2019-11-13 11:00:19.000000000 +0200
+++ cmake/packages/FindGLIB.cmake	2023-11-03 07:49:34.466000000 +0200
@@ -126,7 +126,8 @@
   set_property(TARGET GLIB::GLIB PROPERTY
                INTERFACE_INCLUDE_DIRECTORIES "${GLIB_INCLUDE_DIRS}")
   set_property(TARGET GLIB::GLIB PROPERTY
-               INTERFACE_LINK_LIBRARIES "${GLIB_LIBRARIES}")
+               INTERFACE_LINK_LIBRARIES "${PC_GLIB_LIBRARIES}")
+  set_property(TARGET GLIB::GLIB PROPERTY INTERFACE_LINK_DIRECTORIES "${PC_GLIB_LIBDIR}")
   if(DEFINED GLIB_GIO_LIBRARIES)
     set_property(TARGET GLIB::GLIB APPEND PROPERTY
                  INTERFACE_LINK_LIBRARIES "${GLIB_GIO_LIBRARIES}")

--- CMakeLists.txt 2023-11-03 10:06:29.967000000 +0200
+++ CMakeLists.txt 2023-11-03 10:08:14.996000000 +0200
@@ -109,7 +109,7 @@
 find_package(Gettext REQUIRED)
 find_package_with_target(Intl REQUIRED)
 if(UNIX AND NOT APPLE)
-  find_package(GLIB 2.6 REQUIRED COMPONENTS gio gobject)
+  find_package(GLIB 2.6 REQUIRED COMPONENTS gio gobject gmodule)
 else()
   find_package(GLIB 2.58 REQUIRED COMPONENTS gio gobject)
 endif()
