--- fontforge/CMakeLists.txt	2020-01-27 10:39:51.000000000 +0200
+++ fontforge/CMakeLists.txt	2023-11-03 10:04:11.517000000 +0200
@@ -114,8 +114,6 @@
 )
 
 add_library(fontforge
-  ${FONTFORGE_NOINST_HEADERS}
-  ${FONTFORGE_INST_HEADERS}
   activeinui.c
   asmfpst.c
   autohint.c
@@ -212,6 +210,9 @@
   zapfnomen.c
 )
 
+set_target_properties(fontforge PROPERTIES PUBLIC_HEADER "${FONTFORGE_INST_HEADERS}")
+set_target_properties(fontforge PROPERTIES PRIVATE_HEADER "${FONTFORGE_NOINST_HEADERS}")
+
 if(ENABLE_WOFF2_RESULT)
   target_sources(fontforge PRIVATE woff2.cc)
 endif()
@@ -271,11 +272,14 @@
   target_link_libraries(fontforge PRIVATE gunicode gutils)
 endif()
 
-# No dev package -> no need to install if static
-if(BUILD_SHARED_LIBS)
-  if(WIN32 OR CYGWIN)
-    install(TARGETS fontforge RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
-  else()
-    install(TARGETS fontforge RUNTIME DESTINATION ${CMAKE_INSTALL_LIBDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
-  endif()
+if(WIN32)
+  install(TARGETS fontforge RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+else()
+  install(TARGETS fontforge
+    RUNTIME DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    PRIVATE_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/private/fontforge
+    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/fontforge
+  )
 endif()

--- inc/CMakeLists.txt
+++ inc/CMakeLists.txt
@@ -23,3 +23,35 @@
     Gcov::Gcov
 )
 endif()
+
+set(FONTFORGE_COMMON_INST_HEADERS
+  "${CMAKE_CURRENT_BINARY_DIR}/../inc/fontforge-config.h"
+  "${CMAKE_CURRENT_BINARY_DIR}/../inc/fontforge-version-extras.h"
+  ../inc/basics.h
+  ../inc/carbon.h
+  ../inc/dlist.h
+  ../inc/ffgdk.h
+  ../inc/ffglib.h
+  ../inc/gdraw.h
+  ../inc/gfile.h
+  ../inc/ggadget.h
+  ../inc/gicons.h
+  ../inc/gimage.h
+  ../inc/gkeysym.h
+  ../inc/gprogress.h
+  ../inc/gresedit.h
+  ../inc/gresource.h
+  ../inc/gutils.h
+  ../inc/gwidget.h
+  ../inc/gwwiconv.h
+  ../inc/hotkeys.h
+  ../inc/intl.h
+  ../inc/prefs.h
+  ../inc/ustring.h
+  ../inc/utype.h
+)
+set_target_properties(fontforge_common_headers PROPERTIES PUBLIC_HEADER "${FONTFORGE_COMMON_INST_HEADERS}")
+
+install(TARGETS fontforge_common_headers
+    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/fontforge/inc
+)
