--- gutils/CMakeLists.txt	2019-11-13 11:00:19.000000000 +0200
+++ gutils/CMakeLists.txt	2023-11-03 08:16:14.105000000 +0200
@@ -27,9 +27,7 @@
   prefs.c
 )

-if(BUILD_SHARED_LIBS)
-  set_property(TARGET gutils PROPERTY POSITION_INDEPENDENT_CODE ON)
-endif()
+set_property(TARGET gutils PROPERTY POSITION_INDEPENDENT_CODE ON)

 list(APPEND gutils_LIBRARIES Intl::Intl)
 if(ENABLE_LIBGIF_RESULT)

--- Unicode/CMakeLists.txt	2020-03-03 11:24:02.000000000 +0200
+++ Unicode/CMakeLists.txt	2023-11-03 07:33:25.986000000 +0200
@@ -11,9 +11,7 @@
   utype.c
 )
 
-if(BUILD_SHARED_LIBS)
-  set_property(TARGET gunicode PROPERTY POSITION_INDEPENDENT_CODE ON)
-endif()
+set_property(TARGET gunicode PROPERTY POSITION_INDEPENDENT_CODE ON)
 
 if(${CMAKE_VERSION} VERSION_LESS "3.12.0")
   make_object_interface(gunicode
